contracts/SafeRelayBoundedRefund.sol

```diff
@@ -152,20 +155,27 @@ contract SafeRelayBoundedRefund is BoundaryManager, ReentrancyGuard {
      * @return payment Amount of refunded gas
      */
     function handleRefund(RefundParams calldata refundParams, uint256 startGas) internal returns (uint256 payment) {
+        // COMMENT: maybe good to extract the address(0) to a named constant. 
+
         // solhint-disable-next-line avoid-tx-origin
         address payable receiver = refundParams.refundReceiver == address(0) ? payable(tx.origin) : refundParams.refundReceiver;
 
         uint256 gasConsumed = startGas - gasleft() + COVERED_REFUND_PAYMENT_GAS;
         payment = min(gasConsumed, refundParams.gasLimit) * refundParams.maxFeePerGas;
 
         if (refundParams.gasToken == NATIVE_TOKEN) {
+            // COMMENT: for defensive programming, I would extract the output of the `execute` into a separate variable and 
+            // check the variable in the if statement.
```

Comment:
In solidity this actually creates unnecessary overhead in stack items. This can be resolved by the optimizer, but I would not use the optimizer if not required.

Reply:
I'd actually use the optimizer for this contract because when I tested the gas savings are crazy, like tens of thousands

Reply:
You also told me that the solidity team assumes that with solidity 0.8+, people use the optimizer. I checked, and most of the ecosystem projects use the optimizer as well, so the honey pot amount is crazy

Reply:
> You also told me that the solidity team assumes that with solidity 0.8+

Doesn't mean that it is good :P

The auditors from the vesting contract actually also recommended to NOT use the optimizer, as it is not audited and just adds risk.

The risk to use the optimizer here is probably low, but if we use this contract primarily on L2 and sidechains gas saving is not really important and I would probably disable it.

Reply:
> but if we use this contract primarily on L2 and sidechains gas saving is not really important and I would probably disable it.

Afair the plan is to do a pilot launch on l2/sidechain, but eventually, it should go live on mainnet

Reply:
I am not sure if it is smart to try to use the same contract on all chains with all functionality. I would always try to have only the minimal required logic in a smart contract and separate concerns as much as possible.

The different chains for different when it comes to gas metering and fee payment. The same for the different erc20 tokens. Therefore I would have focused on one of these use cases first (e.g. Native token payment on xDai) and focus the contract on that.